spec/mailers/user_mailer_spec.rb

```diff
@@ -197,4 +197,52 @@
       end
     end
   end
+  
+  describe "invitation" do
+    token = 'abc123'
+
+    before(:each) do
+      @invitation = FactoryGirl.create(:invitation, token: token)
+    end
+
+    let(:email) { UserMailer.invitation(@invitation.id).deliver }
+
+    # Test the headers
+    it 'should have a valid from line' do
+      text = "Archive of Our Own <#{ArchiveConfig.RETURN_ADDRESS}>"
+      expect(email.header['From'].to_s).to eq(text)
+    end
+
+    it 'should have the correct subject line' do
+      text = "[#{ArchiveConfig.APP_SHORT_NAME}] Invitation"
+      expect(email.subject).to eq(text)
+    end
+
+    # Test both body contents
+    it_behaves_like "multipart email"
+
+    describe 'HTML version' do
+      it 'should have text contents' do
+        expect(get_message_part(email, /html/)).to include("like to join us, please sign up at the following address")
+      end
+      
+      it 'should not have missing translations' do
+        expect(get_message_part(email, /html/)).not_to include("translation missing")
+      end
+      
+      it 'should not have exposed HTML' do
+        expect(get_message_part(email, /html/)).not_to include("style")
```

Comment:
I think you'll need to look for something like `&lt;` or `&gt;` here instead of `style`. This seems to be looking at the email's HTML, not the text that's visible to a human reading it, so it should and does find `style`. `&lt;` or similar would indicate the email had a `<` that was visible to the user, which would generally indicate exposed HTML. (It's not foolproof for emails where we show work titles and summaries that might legitimately contain a `<`, but it should work well enough for this invitation email.)